### DISCUSSION TITLE
Building PWA with Mobile template and Web view template

### DISCUSSION THREAD

**Discussion by mattiLeBlanc** [NativeScript/NativeScript#9861] [Q&A]
Hi,

I am looking into NS8 and when I look at the doco's I am not sure if it is still possible to create an Angular app with specific mobile template and a web view template.
Before version 8, there would be a `template. tns.html` flle and a `template.html` file. 
Looking a tutorials and examples, I only see Mobile components.

What if you want to use Angular Material Design for the web version of the dashboard and use the Mobile components for IOS and Android. Is this still possible?

Thanks for pointing me in the right direction.

**Comment by rigor789** [comment]
We have some details about code sharing in the docs here: https://docs.nativescript.org/code-sharing/index.html

Not sure if you've seen that part of the docs - but generally the old approach with `.tns`suffixes had many downfalls and there are better alternatives mentioned above - specifically nx + xplat is one we commonly use.

**Reply by callmeocean** [reply]
This is very frustrating... I have spent the whole day trying to figure out how to create an app that would also build for web. It used to be possible with version 7: https://v7.docs.nativescript.org/angular/code-sharing/creating-a-new-project. Pretty simple with ng command. And now nowhere does it states that the approach has changed. There's just a new article with bunch of separate tools mentioned, without even mentioning that you abandoned the previous flow.

This is a game changer

**Reply by mattiLeBlanc** [reply]
I agree. That page about code-sharing just list a bunch of tools. What we would like to see is an example app of having 2 different views, one for mobile (native) and one for desktop. This was the strength of native script as I understood it.

Now, I could better just use ReactNative with Material components or something like that.
